tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -103,30 +103,50 @@ steps:
           f=$packageName
         fi
         if [[ $f != "" ]]; then
-          npm publish $f $tag ${{ parameters.publishFlags }} 2>&1 | tee publish_log
-          if grep -q "npm ERR!" "publish_log"
-          then
-              if ! grep -q "You cannot publish over the previously published versions" "publish_log"
-              then
-                let t1+=1
-              else
-                echo Package has already been published.
-                local_f="${f}_local"
-                mv "$f" "$local_f"
-                package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
-                npm pack $package_name
-                if cmp -s "$f" "$local_f"
+          for i in $( seq 1 $maximumRetryIfNetworkError )
+          do
+            npm publish $f $tag ${{ parameters.publishFlags }} 2>&1 | tee publish_log
+            if grep -q "npm ERR!" "publish_log"
+            then
+                if grep -q "code ENOTFOUND" "publish_log"
                 then
-                  echo Continuing as published package matches the current one that was attempting to be released.
+                  if ! $i -eq $maximumRetryIfNetworkError
+                  then
+                    echo "Network Error Detected"
+                    continue
+                  else
+                    echo "Final Network Error"
+                    let t1+=1
+                    exit $t1
+                  fi
                 else
-                  echo ERROR: Published package does not match the current one attempting to be released for the same version.
-                  let t1+=1
+                  if ! grep -q "You cannot publish over the previously published versions" "publish_log"
+                  then
+                    let t1+=1
+                  else
+                    echo Package has already been published.
+                    local_f="${f}_local"
+                    mv "$f" "$local_f"
+                    package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
+                    npm pack $package_name
+                    if cmp -s "$f" "$local_f"
+                    then
+                      echo Continuing as published package matches the current one that was attempting to be released.
+                    else
+                      echo ERROR: Published package does not match the current one attempting to be released for the same version.
+                      let t1+=1
```

Comment:
t1 used be used for tracking how many errors we encountered, but since we are going to exist on error now, we probably don't need it anymore.

Reply:
Removed t1 from the codebase and made exit status to `exit 1` 

As a result, made changes to the existing if-statement

Before
```
if ! grep -q "You cannot publish over the previously published versions" "publish_log"
    then
        let t1+=1
    else
        echo Package has already been published.
        local_f="${f}_local"
        mv "$f" "$local_f"
        package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
        npm pack $package_name
        if cmp -s "$f" "$local_f"
        then
            echo Continuing as published package matches the current one that was attempting to be released.
        else
             echo ERROR: Published package does not match the current one attempting to be released for the same version.
             let t1+=1
         fi
      fi
      rm "$f"
      mv "$local_f" "$f"
      exit $t1
fi
```

After
if grep -q "You cannot publish over the previously published versions" "publish_log"
    then
        echo Package has already been published.
        local_f="${f}_local"
        mv "$f" "$local_f"
        package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
        npm pack $package_name
        if cmp -s "$f" "$local_f"
        then
            echo Continuing as published package matches the current one that was attempting to be released.
        else
             echo ERROR: Published package does not match the current one attempting to be released for the same version.
        fi
     fi
      rm "$f"
      mv "$local_f" "$f"
      exit $t1
  fi